### PR TITLE
Add stylua support

### PR DESCRIPTION
**Pull request by keith**
This adds support for [stylua](https://github.com/JohnnyMorganz/StyLua)

### CODE REVIEW

**Comment by chtenb** [comment]
Looks good. Would you care to update the README?

**Comment by keith** [comment]
Thanks done!

File: plugin/defaults.vim [hunk 1]
```diff
@@ -620,6 +620,9 @@ endif
 if !exists('g:formatdef_luafmt')
     let g:formatdef_luafmt = "'luafmt --stdin '.bufname('%')"
 endif
+if !exists('g:formatdef_stylua')
+    let g:formatdef_stylua = "'stylua --search-parent-directories --stdin-filepath ' . expand('%:p') .' -- -'"
+endif
 if !exists('g:formatters_lua')
     let g:formatters_lua = ['luafmt']
```
Comment (chtenb) [inline plugin/defaults.vim:627]: You forgot to add the formatter in the default list. You probably had it set in your vimrc :)

Comment (keith) [reply plugin/defaults.vim]: ah that was intentional to not change behavior on folks if they were just relying on the default, updated

**Comment by keith** [comment]
thanks!

**Comment by chtenb** [comment]
Thanks

**Pull request merged**